includes/classes/Feature/BooleanSearchOperators.php

```diff
@@ -0,0 +1,339 @@
+			 *
+			 * @return {array} New query
+			 */
+			return \apply_filters( 'ep_labs_boolean_operators_query_args', $simple_query, $query_vars, $search_text, $search_fields, $query );
+		}
+
+		return $query;
+	}
+
+	/**
+	 * Test if a search text uses boolean operators
+	 * reference for this RegEx: https://regex101.com/r/JizB0Z/1
+	 * reference for acceptable operators:
+	 * https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html#simple-query-string-syntax
+	 *
+	 * # matches
+	 * (museum (art))    # no extra requirements for surrounding parentheses
+	 * "american museum" # no extra requirements for surrounding double quotes
+	 * museum art*       # requires zero space before the operator
+	 * museum art~35     # requires zero space before and at least one digit after the operator
+	 * museum | art      # requires surrounding spaces
+	 * museum +art       # requires space before
+	 * museum -art       # requires space before
+	 * museum OR art     # requires surrounding spaces and uppercase operator
+	 * museum AND art    # requires surrounding spaces and uppercase operator
+	 * museum NOT art    # requires surrounding spaces and uppercase operator
+	 *
+	 * # non matches
+	 * museum art
+	 * art ()
+	 * museum art 1985
+	 * 35 artists~
+	 * museum or art
+	 * museum and art
+	 * museum not art
+	 * "museum
+	 * (museum
+	 * museum)
+	 * museum *art
+	 *
+	 * @param string $search_text the search query
+	 *
+	 * @return bool
+	 */
+	public function query_uses_boolean_operators( $search_text ) {
+		$boolean_regex = '/(\".+\")|(\+)|(\-)|(\S\*)|(\S\~\d)|(.+\|.+)|(\(\S+\))|(\sOR\s)|(\sAND\s)|(\sNOT\s)/';
+
+		return (bool) preg_match( $boolean_regex, $search_text );
+	}
+
+	/**
+	 * Returns requirements status of feature
+	 *
+	 * Requires the search feature to be activated
+	 *
+	 * @return FeatureRequirementsStatus
+	 */
+	public function requirements_status() {
+		/** Features Class @var Features $features */
+		$features = Features::factory();
+
+		/** Search Feature @var Feature\Search\Search $search */
+		$search = $features->get_registered_feature( 'search' );
+
+		if ( ! $search->is_active() ) {
+			return new FeatureRequirementsStatus( 2, esc_html__( 'This feature requires the "Post Search" feature to be enabled', 'elasticpress-labs' ) );
+		}
+
+		return parent::requirements_status();
+	}
+
+	/**
+	 * Output feature box summary
+	 */
+	public function output_feature_box_summary() {
+		?>
+		<p><?php esc_html_e( 'Allow boolean operators in search queries', 'elasticpress-labs' ); ?></p>
+		<?php
+	}
+
+	/**
+	 * Output feature box long
+	 */
+	public function output_feature_box_long() {
+		?>
+		<p><?php esc_html_e( 'Allows users to search using the following boolean operators:', 'elasticpress-labs' ); ?></p>
+		<ul>
+			<li>
+				<code>+</code> or
+				<code>AND</code> <?php esc_html_e( 'signifies AND operation. eg.: modern +art, modern AND art', 'elasticpress-labs' ); ?>
```

Comment:
These `or` should be translatable. And to make it 100% translatable, the `<code>+</code>` and the `<code>AND</code>` should be replaced by placeholders, so RTL languages can place them in (our) end of the string. Something like:
```
echo wp_kses_post(
	sprintf(
		/* translators: 1: <code>+</code>, 2: <code>AND</code> */
		esc_html__( '%1$s or %2$s signifies AND operation. eg.: modern +art, modern AND art', 'elasticpress-labs' ),
		'<code>+</code>',
		'<code>AND</code>'
	)
);
```